mitiq/pec/representations/learning.py

```diff
@@ -0,0 +1,86 @@
+# Copyright (C) 2022 Unitary Fund
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+"""Function to calculate parameters for biased noise model via a
+learning-based technique."""
+
+from typing import Optional, List
+import numpy as np
+from cirq import Circuit, LineQubit, Gate
+from mitiq import QPROGRAM, Executor, Observable
+from mitiq.pec import execute_with_pec
+from mitiq.pec.representations.biased_noise import (
+    represent_operation_with_local_biased_noise,
+)
+
+
+def biased_noise_loss_function(
+    params: np.ndarray,
+    operation: Gate,
+    training_circuits: List[QPROGRAM],
+    ideal_values: np.ndarray,
+    noisy_executor: Executor,
```

Comment:
With the current code, it is impossible to set options for `execute_with_pec`, as, for example, the number of pec samples, the accuracy, etc....
This is a quite strong restriction. For example, setting a smaller number of PEC samples can speedup the computation. Not sure what is the best way to solve this problem. Maybe we can add another argument like
`pec_kwargs : Dict["str", Any]`,
such that, in the body of the function, we can call `execute_with_pec( ....  ,  **pec_kwargs)`.

Reply:
Good point! I applied your suggestion to add the argument `pec_kwargs` in the `biased_noise_loss_function`. 

Also I'm now using`pec_kwargs` to set a looser precision in `test_biased_noise_loss_function` and `test_biased_noise_loss_compare_ideal`